connect/transforms/src/main/java/org/apache/kafka/connect/transforms/Cast.java

```diff
@@ -0,0 +1,363 @@
+                    else if (value instanceof Boolean)
+                        return ((boolean) value) ? (long) 1 : (long) 0;
+                    else if (value instanceof String)
+                        return Long.parseLong((String) value);
+                    else
+                        throw new ConnectException("Unexpected type in Cast transformation: " + value.getClass());
+                case FLOAT32:
+                    if (value instanceof Number)
+                        return ((Number) value).floatValue();
+                    else if (value instanceof Boolean)
+                        return ((boolean) value) ? 1.f : 0.f;
+                    else if (value instanceof String)
+                        return Float.parseFloat((String) value);
+                    else
+                        throw new ConnectException("Unexpected type in Cast transformation: " + value.getClass());
+                case FLOAT64:
+                    if (value instanceof Number)
+                        return ((Number) value).doubleValue();
+                    else if (value instanceof Boolean)
+                        return ((boolean) value) ? 1. : 0.;
+                    else if (value instanceof String)
+                        return Double.parseDouble((String) value);
+                    else
+                        throw new ConnectException("Unexpected type in Cast transformation: " + value.getClass());
+                case BOOLEAN:
+                    if (value instanceof Number)
+                        return ((Number) value).longValue() > 0L;
+                    else if (value instanceof Boolean)
+                        return value;
+                    else if (value instanceof String)
+                        return Boolean.parseBoolean((String) value);
+                    else
+                        throw new ConnectException("Unexpected type in Cast transformation: " + value.getClass());
+                case STRING:
+                    return value.toString();
+                default:
+                    throw new ConnectException(targetType.toString() + " is not supported in the Cast transformation.");
+            }
+        } catch (NumberFormatException e) {
+            throw new ConnectException("Value (" + value.toString() + ") was out of range for requested data type", e);
+        }
+    }
+
+    @Override
+    public ConfigDef config() {
+        return CONFIG_DEF;
+    }
+
+    @Override
+    public void close() {
+    }
+
+    protected abstract Schema operatingSchema(R record);
+
+    protected abstract Object operatingValue(R record);
+
+    protected abstract R newRecord(R record, Schema updatedSchema, Object updatedValue);
+
+    private static Map<String, Schema.Type> parseFieldTypes(List<String> mappings) {
+        final Map<String, Schema.Type> m = new HashMap<>();
+        for (String mapping : mappings) {
+            final String[] parts = mapping.split(":");
```

Comment:
Not trimming the parts after split, feels a bit restrictive. Do we want to use a util method that does both? 
Change shouldn't have big impact because I see it's only used here and in `ReplaceField` transform.